### ISSUE TITLE
PDO rowCount compatibility problem

### ISSUE THREAD

**Issue by phithi92** [PHPAuth/PHPAuth#517]
Hey,

i found some compatibility problems with databases and PDO rowCount function.

> If the last SQL statement executed by the associated PDOStatement was a SELECT statement, some databases may return the number of rows returned by that statement. However, this behaviour is not guaranteed for all databases and should not be relied on for portable applications. 

https://www.php.net/manual/de/pdostatement.rowcount.php

As Example:
https://github.com/PHPAuth/PHPAuth/blob/04192038e65b59e9f4a4e6e877d34dd8e0644682/Auth.php#L430-L436

In my case, user cant login cause db response allways 0.

my solution for this example:

```
        $query_prepared->execute(['email' => strtolower($email)]);
        
        $result = $query_prepared->fetchColumn();
        
        if( TRUE === empty($result)){
            return 0;
        }

        return $result;
```

**Comment by KarelWintersky** [comment]
your DB?

**Comment by louis123562** [comment]
It depends on the dbh-object, that you created. Maybe we should add an requirement for this in the Auth-Class-Contructor.

**Comment by louis123562** [comment]
Can you show the code of you dbh-object initialization?

**Comment by KarelWintersky** [comment]
it a bug, really.

**Comment by louis123562** [comment]
It depends on with what attributes you initialized it. There is

`PDO::MYSQL_ATTR_FOUND_ROWS => true`

for example.

**Comment by phithi92** [comment]
MySQL 5.7.26
PHP 7.3.9

`
        $options = [
            PDO::ATTR_ERRMODE => self::ERRMODE_EXCEPTION,
            PDO::MYSQL_ATTR_INIT_COMMAND => "SET SESSION sql_mode = 'TRADITIONAL'",
            PDO::MYSQL_ATTR_USE_BUFFERED_QUERY => false,
            PDO::MYSQL_ATTR_FOUND_ROWS => true
        ];

        $PDO = new PDO("mysql:host=localhost;dbname=pi-shop;charset=utf8mb4;port=3306", USER, PASS, $options );

        $container->add(Config::class)->addArgument($PDO);
        $container->add(Auth::class)->addArgument($PDO)->addArgument($container->get(Config::class));
        
        $auth = $container->get(Auth::class);
        $response = $container->get(Response::class);
        
        if('block' === $auth->isBlocked() ){
            
            $container->add(Base_Controller::class);
            return $container->get(Base_Controller::class)->fatalErrorPage($request);
        }
                                
        $result = $auth->login($userPost['username'],$userPost['password'],($userPost['remember'] ? 1:0));

`

**Comment by phithi92** [comment]
> It depends on with what attributes you initialized it. There is
> 
> `PDO::MYSQL_ATTR_FOUND_ROWS => true`
> 
> for example.

found that solution too, but no effect

**Comment by louis123562** [comment]
I dont use it for the phpAuth; it was just a suggestion. What database are you running? Mine runs on MySQL and MariaDB and it works fine calling getUID("email-address"). Are you sure `username` ist the email in your case?!

**Comment by phithi92** [comment]
> I dont use it for the phpAuth; it was just a suggestion. What database are you running? Mine runs on MySQL and MariaDB and it works fine calling getUID("email-address"). Are you sure `username` ist the email in your case?!

im sure that user input is correct. i copy manual data as variable in code to check myself. same result. 

MySQL 5.7.26
PHP 7.3.9
OSX 11.5.2

database entry exist and is equal. with my solution i get the except answer

**Comment by louis123562** [comment]
i think there was an php/pdo error on lower version; can you upgrade your php version to latest, like 7.4.23?

**Comment by phithi92** [comment]
Same on 7.4.1 and 7.4.2

this bug isnt new. i see this first time before 5 years

**Comment by louis123562** [comment]
7.4.1 and 7.4.1 are old versions...

**Comment by KarelWintersky** [comment]
@louis123562 , our minimal compatibility is PHP 7.1 , so, we must support it. 

This issue is really bug of PHPAuth, not an user mistake

**Comment by KarelWintersky** [comment]
Fixed at 1.3.1 release, please validate

**Comment by phithi92** [comment]
it worked like a charm. thanks a lot.
